### PR TITLE
[Math] Fix typo in generated code

### PR DESCRIPTION
**Pull request by dpiparo**
fixes ROOT-7801
https://its.cern.ch/jira/browse/ROOT-7801

### CODE REVIEW

File: hist/hist/src/TMultiDimFit.cxx [hunk 1]
```diff
@@ -2120,8 +2120,8 @@ void TMultiDimFit::MakeRealCode(const char *filename,
    << "        for (k = 3; k <= power; k++) { " << std::endl
    << "          p3 = p2 * v;" << std::endl;
    if (fPolyType == kLegendre)
-      outFile << "          p3 = ((2 * i - 3) * p2 * v - (i - 2) * p1)"
-      << " / (i - 1);" << std::endl;
+      outFile << "          p3 = ((2 * k - 3) * p2 * v - (k - 2) * p1)"
+      << " / (k - 1);" << std::endl;
    if (fPolyType == kChebyshev)
```
Comment (lmoneta) [inline hist/hist/src/TMultiDimFit.cxx:2125]: Are we sure in case of Chebyshev the code is correct? 
There is there no dependency on `k`

**Comment by dpiparo** [comment]
Isn't `k` the index of the for loop above https://github.com/root-project/root/pull/14945/files#diff-cac5cf3aab4459046881d0ce963449104c7c685a469e2d6a590efb911c271ea3R2120 ?

File: hist/hist/src/TMultiDimFit.cxx (hunk 1 above)
Comment (ferdymercury) [reply hist/hist/src/TMultiDimFit.cxx:2125]: In the recurrent Chebyshev formulation, there is no need to specify `k`, the loop depth encodes it:

image

In contrast, for Legendre, you need `k` explicitly:

image

**Review by ferdymercury — APPROVED** [review]

File: hist/hist/src/TMultiDimFit.cxx (hunk 1 above)
Comment (lmoneta) [reply hist/hist/src/TMultiDimFit.cxx:2125]: Thank you @ferdymercury  for checking this!
I agree, the code looks correct now!

**Review by lmoneta — APPROVED** [review]

**Pull request merged**